packages/jsii/lib/symbol-id.ts

```diff
@@ -61,11 +71,27 @@ function assemblyRelativeSourceFile(sourceFileName: string) {
     fs.readFileSync(packageJsonLocation).toString(),
   );
 
-  const sourcePath = removePrefix(
+  let sourcePath = removePrefix(
     packageJson.jsii?.outdir ?? '',
     path.relative(path.dirname(packageJsonLocation), sourceFileName),
   );
 
+  // Modify the namespace if we allow changes and if the outDir exists.
+  // We should only allow changes if we call the function from rosetta.
+  if (!valueAsIs && packageJson.jsii?.tsc?.outDir) {
+    const paths = path.normalize(sourcePath).split(path.sep);
+    const pathDir = paths.shift();
```

Comment:
Looks like we're only comparing the very first path segment. What if `outDir` or `rootDir` consists of 2 path segments? 

```
{
  "rootDir": "src/public",
  "outDir": "dist/minified",
}
```
?